File: competitive-events-api/src/main/java/es/jocasolo/competitiveeventsapi/dto/event/EventPageDTO.java
package es.jocasolo.competitiveeventsapi.dto.event;

import java.io.Serializable;
import java.util.List;

import es.jocasolo.competitiveeventsapi.dto.DTO;

public class EventPageDTO extends DTO implements Serializable {

	private static final long serialVersionUID = 1L;

	private Long total;
	private Integer pages;
	private Boolean hasNext;
	private Boolean hasPrevious;
	private List<EventDTO> events;

	public Long getTotal() {
		return total;
	}

	public void setTotal(Long total) {
		this.total = total;
	}

	public Boolean getHasNext() {
		return hasNext;
	}

	public void setHasNext(Boolean hasNext) {
		this.hasNext = hasNext;
	}

	public Boolean getHasPrevious() {
		return hasPrevious;
	}

	public void setHasPrevious(Boolean hasPrevious) {
		this.hasPrevious = hasPrevious;
	}

	public List<EventDTO> getEvents() {
		return events;
	}

	public void setEvents(List<EventDTO> events) {
		this.events = events;
	}

	public Integer getPages() {
		return pages;
	}

	public void setPages(Integer pages) {
		this.pages = pages;
	}

	@Override
	public String toString() {
		return "EventPageDTO [total=" + total + "]";
	}

}

File: competitive-events-app/app/src/main/java/es/jocasolo/competitiveeventsapp/dto/event/EventPageDTO.kt
package es.jocasolo.competitiveeventsapp.dto.event

import com.google.gson.annotations.SerializedName

class EventPageDTO(
        @SerializedName("total") var total : Long? = null,
        @SerializedName("pages") var pages: Int? = null,
        @SerializedName("hasNext") var hasNext: Boolean? = null,
        @SerializedName("hasPrevious") var hasPrevious: Boolean? = null,
        @SerializedName("events") var events: List<EventDTO>? = null
)